core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/LDAPSearchSpecification.java

```diff
@@ -88,10 +88,14 @@ public LDAPSearchSpecification(RealmConfiguration realmConfig, List<ExpressionCo
         this.searchControls.setSearchScope(SearchControls.SUBTREE_SCOPE);
 
         for (ExpressionCondition expressionCondition : expressionConditions) {
-            if (ExpressionAttribute.ROLE.toString().equals(expressionCondition.getAttributeName())
-                    && isGroupFiltering) {
-                isMultiGroupFiltering = true;
-            } else if (ExpressionAttribute.ROLE.toString().equals(expressionCondition.getAttributeName())) {
+            if (ExpressionAttribute.ROLE.toString().equals(expressionCondition.getAttributeName())) {
+                if (ExpressionOperation.NE.toString().equals(expressionCondition.getOperation())) {
+                    throw new UserStoreClientException(
+                            "ne operation is not supported for group-based user listing in LDAP userstores.");
+                }
+                if (isGroupFiltering) {
+                    isMultiGroupFiltering = true;
+                }
```

Comment:
Let's discuss why we can't give the support.
Got a suggesion but need to test.
`(&(objectClass=person)(!(memberOf=cn=AdminRole,ou=Groups,dc=example,dc=com)))`

Reply:
Had a discussion in the code review and conclude not to support the group based user filtering for the ldap, due to the current implementation limitation we have.